src/routes/auth/facebook.js

```diff
@@ -0,0 +1,123 @@
+import express  from 'express';
+import passport from 'passport';
+import { Strategy } from 'passport-facebook';
+
+import config   from '../../config';
+import db       from '../../models/db';
+import auth     from '../../middlewares/auth';
+import { finalizeAuth } from './utils';
+import {
+  ApiError,
+  BAD_REQUEST, ERRNO_BAD_REQUEST,
+  FORBIDDEN, ERRNO_FORBIDDEN,
+  UNAUTHORIZED, ERRNO_UNAUTHORIZED,
+} from '../../errors';
+
+const router = express.Router();
+
+const callbackURL =
+  `${config.get('publicHost')}/${config.get('version')}/auth/facebook/callback`;
+
+passport.use(new Strategy(
+  {
+    clientID: config.get('facebook.clientID'),
+    clientSecret: config.get('facebook.clientSecret'),
+    callbackURL,
+    passReqToCallback: true,
+    enableProof: true,
+    state: true,
+  },
+  function(req, accessToken, refreshToken, profile, cb) {
+    db()
+      .then(models => {
+        const { AUTH_PROVIDER, authenticate } = models.Users;
+        return authenticate(
+          AUTH_PROVIDER,
+          {
+            opaqueId: profile.id,
+            provider: 'facebook',
+            ClientKey: req.session.clientKey,
+          }
+        );
+      })
+      .then(
+        userInfo => cb(null, userInfo),
+        err => cb(err)
+      );
+  }
+));
+
+function checkClientExists(req, res, next) {
+  db()
+    .then(({ Clients }) => Clients.findById(req.user.id))
```

Comment:
actually, it seems to be a standard practice in express apps that the result of the authentications is in `req.user`. Also `req.user` can contains the information for clients, users or admin (and later: sensors).

The alternative is that instead of having a `scope` we could either set `req.user` `req.client` `req.admin` and `req.sensor` with the information we currently have in `id`. This looks quite more elegant to me. What do you think ?